src/client/index.ts

```diff
@@ -319,17 +326,21 @@ class Client extends EventEmitter {
    * @param req - Request to send to the server.
    * @returns The response from the server.
    */
-  public async request<R extends BaseRequest, T extends BaseResponse>(
+  public async request<R extends Request, T extends Response>(
```

Comment:
I disagree with this - this choice was made so that people can use other request types more easily, they'll just not be typed as well. If it's typed with `BaseRequest`/`BaseResponse`, if someone wants to make an NFT-related request before we have support for that, they can.

Reply:
If you want to do that, then you will have to add that declaration to the function overloads, and not the implimentation.

Reply:
Added to function overloading